dll/cpl/intl/languages.c

```diff
@@ -1,15 +1,31 @@
 #include "intl.h"
 
 #include <shellapi.h>
+#include <shlobj.h>
 #include <strsafe.h>
 
-/* Is there any Japanese input method? */
-BOOL HasJapaneseIME(VOID)
+/* Where is the uninstaller of "ReactOS JPN Package"? */
+BOOL GetJapaneseUninstallPath(HWND hwnd, LPWSTR pszPath, SIZE_T cchPath)
 {
-    WCHAR szImePath[MAX_PATH];
-    GetSystemDirectoryW(szImePath, _countof(szImePath));
-    StringCchCatW(szImePath, _countof(szImePath), L"\\mzimeja.ime");
-    return GetFileAttributesW(szImePath) != INVALID_FILE_ATTRIBUTES;
+    SHGetSpecialFolderPathW(hwnd, pszPath, CSIDL_PROGRAM_FILES, FALSE);
+    StringCchCatW(pszPath, cchPath, L"\\ReactOS JPN Package\\unins000.exe");
+    if (GetFileAttributesW(pszPath) != INVALID_FILE_ATTRIBUTES)
+        return TRUE;
+
+    SHGetSpecialFolderPathW(hwnd, pszPath, CSIDL_PROGRAM_FILESX86, FALSE);
+    StringCchCatW(pszPath, cchPath, L"\\ReactOS JPN Package\\unins000.exe");
+    if (GetFileAttributesW(pszPath) != INVALID_FILE_ATTRIBUTES)
+        return TRUE;
```

Comment:
Maybe use `Uninstall` registry key instead of hardcoding the path. Because JPN package in fact allows to specify different installation path:

![VirtualBox_ReactOS_01_11_2022_15_57_23](https://user-images.githubusercontent.com/578406/199238112-c51f360e-afc2-4077-ba84-a1fa0e45dd04.png)
![VirtualBox_ReactOS_01_11_2022_15_59_41](https://user-images.githubusercontent.com/578406/199238127-61dfd1bf-6be2-49c4-a2a6-a94150bba1c9.png)

Reply:
Fixed.